### PR TITLE
Add deployment package to framework

### PR DESCRIPTION
**Pull request by indyfree**
Add reusable function to check if deployment properties are correct.

### CODE REVIEW

File: pkg/framework/deployment/deployment.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+	}
+
+	return d, nil
+}
+
+// Check ensures that key properties of the deployment are correct.
+func (d *Deployment) Check(name string, replicas int, expectedLabels, expectedMatchLabels map[string]string) error {
+	ds, err := d.k8sClient.Apps().Deployments(metav1.NamespaceSystem).Get(name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return microerror.Newf("could not find deployment: '%s' %v", name, err)
```
Comment (xh3b4sd) [inline pkg/framework/deployment/deployment.go]: Eh, no. See https://github.com/giantswarm/fmt/blob/master/go/errors.md.

**Review by xh3b4sd — CHANGES_REQUESTED** [review]

File: pkg/framework/deployment/deployment.go (hunk 1 above)
Comment (indyfree) [reply pkg/framework/deployment/deployment.go]: Ahh :sweat_smile: too fast.

File: pkg/framework/deployment/deployment.go [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+	"reflect"
+
+	"github.com/giantswarm/microerror"
+	"github.com/giantswarm/micrologger"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+)
+
+type DeploymentConfig struct {
```
Comment (MarcelMue) [inline pkg/framework/deployment/deployment.go]: Can be removed to `Config`.

File: pkg/framework/deployment/deployment.go (hunk 1 above)
Comment (indyfree) [reply pkg/framework/deployment/deployment.go]: The way to go here is to create a custom error matcher i.e. `IsNotFoundError` and wrap/match the apierror there, right? Similar to the promtheus example?

Comment (xh3b4sd) [reply pkg/framework/deployment/deployment.go]: Maybe. Depends on what you want to do. When the error here has to be managed in a special way then we should have control over it and return our own proper type.

Comment (MarcelMue) [reply pkg/framework/deployment/deployment.go]: You could just use `microerror.Maskf` IMO. I am kind of wondering why there is a special case for `NotFound` to begin with. We don't do anything different with it, right?

Comment (indyfree) [reply pkg/framework/deployment/deployment.go]: Decided to go for an own `notFoundError` type, so consuming packages can match it more easily. Also created and masked errors for the other cases. Is that done right?

Comment (xh3b4sd) [reply pkg/framework/deployment/deployment.go]: Do we want to act differently though? My question would be why not one `incorrectDeploymentError` to rule them all? Just thinking loud here.

File: pkg/framework/deployment/deployment.go [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+	}
+
+	return d, nil
+}
+
+// Check ensures that key properties of the deployment are correct.
+func (d *Deployment) Check(name string, replicas int, expectedLabels, expectedMatchLabels map[string]string) error {
+	ds, err := d.k8sClient.Apps().Deployments(metav1.NamespaceSystem).Get(name, metav1.GetOptions{})
+	if apierrors.IsNotFound(err) {
+		return microerror.Maskf(notFoundError, "could not find deployment: '%s'", name)
```
Comment (kopiczko) [inline pkg/framework/deployment/deployment.go]: This error is printed like: `not found: could not find deployment: '...'`. Let's not repeat this information.`not found: deployment: '...'` is enough.

**Review by TheoBrigitte — APPROVED** [review]
LGTM

File: pkg/framework/deployment/deployment.go (hunk 3 above)
Comment (indyfree) [reply pkg/framework/deployment/deployment.go]: Good catch, done.

File: pkg/framework/deployment/deployment.go (hunk 1 above)
Comment (indyfree) [reply pkg/framework/deployment/deployment.go]: You have a good point `incorrectDeploymentError` is good to caputure all incorrect properties, be it labels or replicas. Still IMO `notFoundError` is semantically something different than `incorrect`, and can also cause different actions.

**Review by xh3b4sd — APPROVED** [review]
Not blocking here.

**Review by MarcelMue — APPROVED** [review]
Fine with me :+1:

**Pull request merged**